anesthetic/samples.py

```diff
@@ -645,35 +653,43 @@ def dlogX(self, nsamples=None):
         else:
             return WeightedDataFrame(dlogX, self.index, weights=self.weights)
 
-    def importance_sample(self, logL_new, action='add'):
+    def importance_sample(self, logL_new, action='add', inplace=False):
         """Perform importance re-weighting on the log-likelihood.
 
         Parameters
         ----------
         logL_new: np.array
             New log-likelihood values. Should have the same shape as `logL`.
 
-        action: str
+        action: str, optional
             Can be any of {'add', 'replace', 'mask'}.
                 * add: Add the new `logL_new` to the current `logL`.
                 * replace: Replace the current `logL` with the new `logL_new`.
                 * mask: treat `logL_new` as a boolean mask and only keep the
                         corresponding (True) samples.
             default: 'add'
 
+        inplace: bool, optional
+            Indicates whether to modify the existing array, or return a new
+            frame with importance sampling applied.
+            default: False
+
         Returns
         -------
         samples: NestedSamples
             Importance re-weighted samples.
         """
         samples = merge_nested_samples((self, ))
         samples = super(NestedSamples, samples).importance_sample(
-            logL_new=logL_new, action=action
+            logL_new, action=action
         )
         samples = merge_nested_samples(
             (samples[samples.logL > samples.logL_birth], )
         )
-        return samples
+        if inplace:
+            self.__init__(samples)
```

Comment:
This line is going far back its dynasty to its (great?)grandparents, and I believe it loses some of its `NestedSamples` attributes in the process.

This is partly due to `merge_nested_samples` not taking care of all attributes, and as such already applies to non-inplace samples. But with inplace, we loose a few more (at least in `.tex`).

How do we want `.tex`, `.limits`, `.label`, `.root` to behave under importance re-weighting?

Reply:
Nice spot! This was not expected, and I believe a property of how pandas interprets __init__. I have a replaced with a more verbose method, and with a test that checks all of these are passed on to the object, and that we have a copies of the relevant mutable metadata.